it/controllers/incomeSources/add/UKPropertyBusinessAccountingMethodControllerISpec.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.incomeSources.add
+
+import config.featureswitch.IncomeSources
+import helpers.ComponentSpecBase
+import helpers.servicemocks.IncomeTaxViewChangeStub
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import testConstants.BaseIntegrationTestConstants.testMtditid
+import testConstants.IncomeSourceIntegrationTestConstants.{noPropertyOrBusinessResponse, ukPropertyOnlyResponse}
+
+class UKPropertyBusinessAccountingMethodControllerISpec extends ComponentSpecBase {
+
+  val addUKPropertyBusinessAccountingMethodShowUrl: String = controllers.incomeSources.add.routes.UKPropertyBusinessAccountingMethodController.show().url
+  val addUKPropertyBusinessAccountingMethodSubmitUrl: String = controllers.incomeSources.add.routes.UKPropertyBusinessAccountingMethodController.submit().url
+  val checkUKPropertyDetailsShowUrl: String = controllers.incomeSources.add.routes.CheckUKPropertyDetailsController.show().url
+
+  val continueButtonText: String = messagesAPI("base.continue")
+
+  s"calling GET $addUKPropertyBusinessAccountingMethodShowUrl" should {
+    "render the UK Property Business Accounting Method page" when {
+      "User is authorised" in {
+        Given("I wiremock stub a successful Income Source Details response with no UK properties")
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        When(s"I call GET $addUKPropertyBusinessAccountingMethodShowUrl")
+        val result = IncomeTaxViewChangeFrontend.get("/income-sources/add/uk-property-accounting-method")
+        verifyIncomeSourceDetailsCall(testMtditid)
+
+        result should have(
+          httpStatus(OK),
+          pageTitleIndividual("incomeSources.add.uk-property-business-accounting-method.heading"),
+          elementTextByID("continue-button")(continueButtonText)
+        )
+      }
+    }
+  }
+  s"calling POST $addUKPropertyBusinessAccountingMethodSubmitUrl" should {
+    s"redirect to $checkUKPropertyDetailsShowUrl" when {
```

Comment:
"redirect to $checkUKPropertyDetailsShowUrl" stage is repeated .. can you group it under one stage?